p2p/module_raintree_test.go

```diff
@@ -281,7 +281,8 @@ func generateKeys(_ *testing.T, numValidators int) []cryptoPocket.PrivateKey {
 
 // A mock of the application specific to know if a message was sent to be handled by the application
 // INVESTIGATE(olshansky): Double check that how the expected calls are counted is accurate per the
-//                         expectation with RainTree by comparing with Telemetry after updating specs.
+//
+//	expectation with RainTree by comparing with Telemetry after updating specs.
```

Comment:
Should probably `git checkout main` edits to this file since its out of scope

Reply:
Agreed, but with the occasion, a few words 🙂

I need to edit this file every time I change branch. `gofmt` automatically changes that comment spacing and I included it in the changeset by mistake.

With this I want to say that there's probably something we can improve here in terms of developer experience.

- switching branch and running `make develop_test` is not smooth as you would expect. I need to exclude / comment a line in this file to make it work (`mocksP2P "github.com/pokt-network/pocket/p2p/types/mocks"`)
- making any change to this file and saving it changes the formatting of that comment in many default VSCode setups, mine included. It sounds a bit weird to have to tell every new community dev that they have to ignore the automatic formatting / disable `format on save` and use our special spacing.
   - related: if we want to use automatic linters / formatters this problem will surface there as well because they would do what I did basically (changing the spacing in that comment) so if there's a strong need for a specific formatting, we should enforce it programmatically somehow. 🤔

Reply:
I have some ideas with regarding to this, but haven't fully fleshed them out yet.

# Regard `make develop_test`

The core issue is:
1. We add a new dependency in branch A
2. We sitch to branch B where dependency does not exist
3. We can't `mockgen` because of inconsistent direcotires
4. We can't `go mod tidy` because we need to run `mockgen`

<img width="1608" alt="Screen Shot 2022-09-21 at 8 03 10 PM" src="https://user-images.githubusercontent.com/1892194/191649112-5d7fde32-d0b3-4f19-b31a-b5eacfc314ce.png">

I think something with the `mod build flag` might help (see https://github.com/golang/mock#debugging-errors) , but I haven't quite figured it out yet...

<img width="972" alt="Screen Shot 2022-09-21 at 8 05 26 PM" src="https://user-images.githubusercontent.com/1892194/191649592-78e0230c-53c0-4276-a449-a577191e829d.png">


# Regarding linters

I recently came by https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint, where makrdown linting can be defined via a YAML file. I'm sure something similar exists for Go, so we can make it part of the developer guide to set up their IDE to import these configs.

--- 

Will look into this tomorrow when I need a break from thinking :)

Reply:
Your change to mockgen helped solve one of the problems: https://github.com/pokt-network/pocket/pull/240/files

Reply:
1. @okdas is helping with streamlining configs
2. Created #257 to track this work
3. We resolved the mockgen piece